irksome/dirk_stepper.py

```diff
@@ -101,31 +50,31 @@ def getFormDIRK(F, butch, t, dt, u0, bcs=None):
     stage_F = replace(F, repl)
 
     bcnew = []
-    gblah = []
 
     # For the DIRK case, we need one new BC for each old one (rather
     # than one per stage), but we need a `Function` inside of each BC
     # and a rule for computing that function at each time for each
     # stage.
+    a_vals = numpy.array([MC.Constant(0) for i in range(num_stages)],
+                         dtype=object)
+    d_val = MC.Constant(1.0)
     for bc in bcs:
-        Vbc = bc.function_space()
         bcarg = as_ufl(bc._original_arg)
         bcarg_stage = replace(bcarg, {t: t+c*dt})
-        try:
-            gdat = assemble(interpolate(bcarg, Vbc))
-            gmethod = lambda gd, gc: gd.interpolate(gc)
-        except:  # noqa: E722
-            gdat = assemble(project(bcarg, Vbc))
-            gmethod = lambda gd, gc: gd.project(gc)
+        if bcarg_stage == 0:
```

Comment:
Do you also want to check for zero of the proper shape in case somebody isn’t taking advantage of just doing zero?

Reply:
I am happy to do this, but not UFL-savvy enough to know what the proper way to do so is.  I didn't see a similar check anywhere else in Irksome - am I missing something?